sdk/programmableconnectivity/azure-programmableconnectivity/setup.py

```diff
@@ -0,0 +1,70 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) Python Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+# coding: utf-8
+
+import os
+import re
+from setuptools import setup, find_packages
+
+
+PACKAGE_NAME = "azure-programmableconnectivity"
+PACKAGE_PPRINT_NAME = "Azure Programmableconnectivity"
+
+# a-b-c => a/b/c
+package_folder_path = PACKAGE_NAME.replace("-", "/")
+
+# Version extraction inspired from 'requests'
+with open(os.path.join(package_folder_path, "_version.py"), "r") as fd:
+    version = re.search(r'^VERSION\s*=\s*[\'"]([^\'"]*)[\'"]', fd.read(), re.MULTILINE).group(1)
+
+if not version:
+    raise RuntimeError("Cannot find version information")
+
+
+setup(
+    name=PACKAGE_NAME,
+    version=version,
+    description="Microsoft {} Client Library for Python".format(PACKAGE_PPRINT_NAME),
+    long_description=open("README.md", "r").read(),
+    long_description_content_type="text/markdown",
+    license="MIT License",
+    author="Microsoft Corporation",
+    author_email="azpysdkhelp@microsoft.com",
+    url="https://github.com/Azure/azure-sdk-for-python/tree/main/sdk",
+    keywords="azure, azure sdk",
+    classifiers=[
+        "Development Status :: 4 - Beta",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3 :: Only",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
+        "License :: OSI Approved :: MIT License",
+    ],
+    zip_safe=False,
+    packages=find_packages(
+        exclude=[
+            "tests",
+            # Exclude packages that will be covered by PEP420 or nspkg
+            "azure",
+        ]
+    ),
+    include_package_data=True,
+    package_data={
+        "azure.programmableconnectivity": ["py.typed"],
+    },
+    install_requires=[
+        "isodate<1.0.0,>=0.6.1",
```

Comment:
Please don't set upper bound for the package versions.

Reply:
This wasn't something that I put in manually, it would have been something that the autogenerator added for me.

Should I still remove?

Reply:
Yes, please.

BTW, could you share how you generate the code?

I will take a look to fix the generator too. :)

Reply:
Took the following steps to generate the SDK. Some of these steps have some confirmation popups, I always press `y` on those.

```
npm install -g @azure-tools/typespec-client-generator-cli@0.4.1
mkdir azure_stuff_python_v5
cd azure_stuff_python_v5
git clone git@github.com:aronhegedusms/azure-sdk-for-python.git
cd azure-sdk-for-python
tsp-client init -c  https://github.com/Azure/azure-rest-api-specs/blob/7f00999aba7c2ea14163ccf19f18aa7da30abcce/specification/programmableconnectivity/Azure.ProgrammableConnectivity/tspconfig.yaml #31294 Azure/azure-rest-api-specs
cd sdk/programmableconnectivity/azure-programmableconnectivity
python3 -m venv venv
source venv/bin/activate
python -m pip install --upgrade pip
pip install "tox<5"
tox run -e generate -c ../../../eng/tox/tox.ini --root .
pip install -r dev_requirements.txt
pip install -e .
```

Reply:
And I got these steps from trial and error/ this doc: https://github.com/Azure/azure-sdk-for-python/wiki/Dataplane-Codegen-Quick-Start